### PR TITLE
tinydtls: fix for build system fix

### PR DESCRIPTION
**Pull request by miri64**
Do not merge until https://github.com/rfuentess/TinyDTLS/issues/1 was resolved and the PKG_VERSION of tinydtls updated accordingly in this PR!

### CODE REVIEW

**Comment by miri64** [comment]
Fixes #6036 (I hope)

File: examples/dtls-echo/Makefile [hunk 1]
```diff
@@ -49,6 +49,8 @@ CFLAGS += -DDEVELHELP
 # NOTE: Add the package for TinyDTLS
 USEPKG += tinydtls
 
+USEMODULE += tinydtls
```
Comment (OlegHahm) [inline examples/dtls-echo/Makefile]: A module **and** a package with the same name?

Comment (miri64) [inline examples/dtls-echo/Makefile]: Yes, as far as I understood #5891 the idea now:

* `USEPKG` just is responsible for *getting* the package, while
* `USEMODULE` is responsible for *building* it as a module.

Comment (miri64) [inline examples/dtls-echo/Makefile]: (this is not properly cleaned up in master, so we can't use it as an example ;-))

**Comment by miri64** [comment]
Wait the branch you referenced doesn't provide the patch...

**Comment by miri64** [comment]
Updated board blacklist according to @rfuentess reporting (and also adapted memory unsufficient list accordingly ;-))

File: examples/dtls-echo/Makefile (hunk 1 above)
Comment (OlegHahm) [inline examples/dtls-echo/Makefile]: Why is this needed for this package, but not for other packages (e.g., ccn-lite) then?

Comment (miri64) [inline examples/dtls-echo/Makefile]: Because this package already has a `pkg/$(PKG_NAME)/Makefile.dep`.

**Comment by miri64** [comment]
Of course I forgot to add the Makefile.dep I was talking about in https://github.com/RIOT-OS/RIOT/pull/6053#discussion_r86392900. @rfuentess this fixes the issue we talked about on IRC.

File: examples/dtls-echo/Makefile (hunk 1 above)
Comment (kaspar030) [inline examples/dtls-echo/Makefile]: hm, usepkg is enough, but always necessary to even consider the pkg dir. the usemodule should be redundant here. does this build without ?

Comment (miri64) [inline examples/dtls-echo/Makefile]: I can do it as such

Comment (miri64) [inline examples/dtls-echo/Makefile]: Done

**Comment by miri64** [comment]
Let's try building this a few times to make sure #6036 is really fixed.

**Comment by miri64** [comment]
success no 2

**Comment by miri64** [comment]
success no 3

**Comment by rfuentess** [comment]
Meanwhile murdock is running, I'm testing directly on my local copy.  Right now,  17 successful testing of `make distclean; make buildtest`

**Comment by miri64** [comment]
I will do five runs of Murdock (minus the one after rebase).

**Comment by rfuentess** [comment]
As long Murdock is OK  I think  the changes are good, so, ACK !

**Comment by miri64** [comment]
success no 4

**Comment by miri64** [comment]
success no 5 \o/ rebasing now.

**Comment by miri64** [comment]
Done

**Comment by miri64** [comment]
Backport provided in #6061.

**Comment by miri64** [comment]
Murdock fail weird but unrelated (for once ^^)

**Comment by miri64** [comment]
(just FYI, it was #5848)

File: examples/dtls-echo/Makefile (hunk 1 above)
Comment (OlegHahm) [inline examples/dtls-echo/Makefile]: Supongo que es muy útil de abrir los ojos para postear un link. :wink:

Comment (miri64) [inline examples/dtls-echo/Makefile]: Fixed :P
